solana-programs/anchor/audius-data/tests/test-helpers.ts

```diff
@@ -445,6 +470,141 @@ export const testUpdatePlaylist = async ({
   expect(accountPubKeys[2]).to.equal(userAuthorityKeypair.publicKey.toString());
 };
 
+export const testCreateUserDelegate = async ({
+  adminKeypair,
+  adminStorageKeypair,
+  program,
+  provider,
+  replicaSet,
+  replicaSetBumps,
+  cn1,
+  cn2,
+  cn3,
+}) => {
+  const { ethAccount, handleBytesArray, metadata, userId } =
+    initTestConstants();
+  const {
+    baseAuthorityAccount,
+    bumpSeed: userBumpSeed,
+    derivedAddress: userAccountPDA,
+  } = await findDerivedPair(
+    program.programId,
+    adminStorageKeypair.publicKey,
+    Buffer.from(handleBytesArray)
+  );
+
+  // disable admin writes
+  await updateAdmin({
+    program,
+    isWriteEnabled: false,
+    adminStorageAccount: adminStorageKeypair.publicKey,
+    adminAuthorityKeypair: adminKeypair,
+  });
+
+  // New sol key that will be used to permission user updates
+  const newUserKeypair = anchor.web3.Keypair.generate();
+
+  // Generate signed SECP instruction
+  // Message as the incoming public key
+  const message = newUserKeypair.publicKey.toBytes();
+
+  await testCreateUser({
+    provider,
+    program,
+    message,
+    ethAccount,
+    baseAuthorityAccount,
+    handleBytesArray,
+    bumpSeed: userBumpSeed,
+    metadata,
+    newUserKeypair,
+    userStorageAccount: userAccountPDA,
+    adminStoragePublicKey: adminStorageKeypair.publicKey,
+    replicaSet,
+    replicaSetBumps,
+    cn1,
+    cn2,
+    cn3,
+    userId,
+  });
+
+  // Init AuthorityDelegationStatus for a new authority
+  const userAuthorityDelegateKeypair = anchor.web3.Keypair.generate();
+  const authorityDelegationStatusSeeds = [
+    Buffer.from("authority-delegation-status", "utf8"),
+    userAuthorityDelegateKeypair.publicKey.toBytes().slice(0, 32),
+  ];
+
+  const authorityDelegationStatusRes = await PublicKey.findProgramAddress(
+    authorityDelegationStatusSeeds,
+    program.programId
+  );
+  const authorityDelegationStatusPDA = authorityDelegationStatusRes[0];
+  const authorityDelegationStatusBump = authorityDelegationStatusRes[1];
+
+  const initAuthorityDelegationStatusArgs = {
+    accounts: {
+      delegateAuthority: userAuthorityDelegateKeypair.publicKey,
+      authorityDelegationStatusPda: authorityDelegationStatusPDA,
+      payer: provider.wallet.publicKey,
+      systemProgram: SystemProgram.programId,
+    },
+    signers: [userAuthorityDelegateKeypair],
+  };
+
+  await program.rpc.initAuthorityDelegationStatus(
```

Comment:
can we go ahead and verify the decoded instruction for these as well? just to have that coverage